pkg/distributor/ha_tracker.go

```diff
@@ -113,16 +120,24 @@ func (cfg *HATrackerConfig) RegisterFlags(f *flag.FlagSet) {
 func newClusterTracker(cfg HATrackerConfig) (*haTracker, error) {
 	codec := codec.Proto{Factory: ProtoReplicaDescFactory}
 
-	if cfg.FailoverTimeout <= cfg.UpdateTimeout {
-		return nil, fmt.Errorf("HA Tracker failover timeout must be greater than update timeout, %d is <= %d", cfg.FailoverTimeout, cfg.UpdateTimeout)
+	var jitter time.Duration
+	if cfg.UpdateTimeoutJitterMax != 0 {
+		jitter = time.Duration(rand.Int63n(int64(2*cfg.UpdateTimeoutJitterMax))) - cfg.UpdateTimeoutJitterMax
 	}
+
+	if cfg.FailoverTimeout <= cfg.UpdateTimeout+cfg.UpdateTimeoutJitterMax+time.Second {
```

Comment:
This is not the right place to validate the config. Please add `func (cfg *HATrackerConfig) Validate() error` and move the validation there (should return an error on validation failure, `nil` otherwise), add the `Validate()` function to `distributor.Config` (this should call `HATrackerConfig.Validate()`) and the call the `distributor.Config.Validate()` method from the `Validate()` function in `cortex.go` (looks complex, but it's a very easy chain of `Validate()` calls).

Reply:
im okay with making this change - however wondering if it would make more sense to abstract this out as we add more validations. right now it is only validating a single field. but yea, lmk.

Reply:
I would get stick to the PR's scope. Config validation support has been introduced very recently and will be adopted more over the time. My suggestion is to validate only the new config option introduced with this PR, while we'll improve the rest of config over the time.